code_search/app/code_search/nmslib/search_engine.py

```diff
@@ -0,0 +1,69 @@
+import json
+import requests
+import nmslib
+import numpy as np
+from tensor2tensor import problems  # pylint: disable=unused-import
+from code_search.t2t.query import get_encoder_decoder, encode_query
+
+
+class CodeSearchEngine:
+  """This is a utility class which takes an nmslib
+  index file and a data file to return data from"""
```

Comment:
What is in the data file?

Reply:
This part is only representative but data file is meant for a reverse lookup of human-readable code corresponding to the returned indices from nmslib NN search.